File: doc-examples/nitrite-example-java/src/main/java/example/ProjectRepositoryExample.java
package example;

import jakarta.inject.Inject;

/**
 * This class exists to provide documentation snippets (it is not executed as a test).
 */
final class ProjectRepositoryExample {

    @Inject ProjectRepository repository;

    // tag::useEmbeddedId[]
    void useEmbeddedId() {
        ProjectId id = new ProjectId(10, 20);
        repository.save(new Project(id, "Alpha"));
        repository.findById(id);
        repository.deleteById(id);
    }
    // end::useEmbeddedId[]
}

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/ProjectRepositoryExample.kt
package example

import jakarta.inject.Inject

/**
 * This class exists to provide documentation snippets (it is not executed as a test).
 */
class ProjectRepositoryExample {

    @Inject lateinit var repository: ProjectRepository

    fun useEmbeddedId() {
        val id = ProjectId(10, 20)
        repository.save(Project(id, "Alpha"))
        repository.findById(id)
        repository.deleteById(id)
    }
}

